src/trainer.py

```diff
@@ -96,22 +102,20 @@ def build_trainer(params, model, run_dir, metric_should_decrease=True):
                               'verbose': True})
         log.info('\tUsing ReduceLROnPlateau scheduler!')
 
-    train_params = Params(
-        {
-            'cuda_device': params['cuda'],
-            'patience': params['patience'],
-            'grad_norm': params['max_grad_norm'],
-            'val_interval': params['val_interval'],
-            'max_vals': params['max_vals'],
-            'lr_decay': .99,
-            'min_lr': params['min_lr'],
-            'keep_all_checkpoints': params['keep_all_checkpoints'],
-            'val_data_limit': params['val_data_limit'],
-            'max_epochs': params['max_epochs'],
-            'dec_val_scale': params['dec_val_scale'],
-            'training_data_fraction': params['training_data_fraction']})
-    trainer = SamplingMultiTaskTrainer.from_params(model, run_dir,
-                                                   copy.deepcopy(train_params))
+    train_params = Params({'cuda_device': params['cuda'],
+                           'patience': params['patience'],
+                           'grad_norm': params['max_grad_norm'],
+                           'val_interval': params['val_interval'],
+                           'max_vals': params['max_vals'],
+                           'lr_decay': .99, 'min_lr': params['min_lr'],
+                           'keep_all_checkpoints': params['keep_all_checkpoints'],
+                           'val_data_limit': params['val_data_limit'],
+                           'max_epochs': params['max_epochs'],
+                           'dec_val_scale': params['dec_val_scale'],
+                           'training_data_fraction': params['training_data_fraction']})
+    if train_type == "SamplingMultiTaskTrainer":
+        trainer = SamplingMultiTaskTrainer.from_params(model, run_dir,
+                                                       copy.deepcopy(train_params))
     return trainer, train_params, opt_params, schd_params
```

Comment:
What value does trainer have if train_type isn't SamplingMultiTaskTrainer? If it's supposed to fail, make this an assert earlier on.